spack/package.py

```diff
@@ -102,6 +101,10 @@ class Oommf(Package):
     depends_on("tcl", type=("build", "test", "run"))
     depends_on("xproto", type=("build"))
 
+    # Compilation with clang does not work yet (gcc works fine, nothing else tested)
+    # (https://github.com/spack/spack/pull/26933#pullrequestreview-789754233)
+    conflicts("%clang")
```

Comment:
OOMMF should have some clang support: https://github.com/fangohr/oommf/blob/master/oommf/config/platforms/clang-support.tcl
Maybe some changes are required somewhere in here?: https://github.com/fangohr/oommf/blob/67fa0d69eadbbb9eef320babd07910f6d7b4e089/oommf/config/platforms/linux-x86_64.tcl#L100
I think exporting the correct `OOMMF_CPP` environment variable could be sufficient: https://github.com/fangohr/oommf/blob/67fa0d69eadbbb9eef320babd07910f6d7b4e089/oommf/config/platforms/linux-x86_64.tcl#L43 (this takes precedence over L100 above)

Reply:
Nice. I'll create a new issue for this to get to some closure with the MR and gcc soon.